dotnet/src/SemanticKernel.Core/Extensions/KernelSemanticFunctionExtensions.cs

```diff
@@ -110,24 +112,27 @@ public static class KernelSemanticFunctionExtensions
     /// <param name="functionName">A name for the given function. The name can be referenced in templates and used by the pipeline planner.</param>
     /// <param name="pluginName">An optional plugin name, e.g. to namespace functions with the same name. When empty,
     /// the function is added to the global namespace, overwriting functions with the same name</param>
+    /// <param name="promptTemplateFactory">Prompt template factory</param>
     /// <returns>A function ready to use</returns>
     public static ISKFunction CreateSemanticFunction(
         this IKernel kernel,
         string promptTemplate,
         PromptTemplateConfig promptTemplateConfig,
         string? functionName = null,
-        string? pluginName = null)
+        string? pluginName = null,
+        IPromptTemplateFactory? promptTemplateFactory = null)
     {
         functionName ??= RandomFunctionName();
         Verify.ValidFunctionName(functionName);
         if (!string.IsNullOrEmpty(pluginName)) { Verify.ValidPluginName(pluginName); }
 
-        var template = new PromptTemplate(promptTemplate, promptTemplateConfig, kernel.PromptTemplateEngine);
+        var factory = promptTemplateFactory ?? CreateDefaultPromptTemplateFactory(kernel);
+        IPromptTemplate promptTemplateInstance = factory.CreatePromptTemplate(promptTemplate, promptTemplateConfig);
```

Comment:
Just as an observation, no action required: As far as I understand it, the reason why the variable `promptTemplateInstance` is named the way it is, instead of being named `promptTemplate`, is to avoid a name collision between the parameter and the variable names. I think the root cause of it lies in the name of the class we decided to use to encapsulate prompt template rendering functionality - 'PromptTemplate.' The class is not strictly a prompt template, even though it accepts a prompt template in its constructor. It functions more as a prompt template rendering engine that accepts a prompt and can render it. So, I think the name 'PromptTemplateEngine' would be more appropriate for the class, as it better communicates what this class does and is responsible for.